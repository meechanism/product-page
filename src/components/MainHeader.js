import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import Colors from '../styles/colors';
import MediaQuery from '../styles/media-queries';

export const Branding = styled.span`
  display: inline-block;
  font-weight: bold;
  font-size: 3rem;
  border-left: 1rem solid ${Colors.secondary};
  padding-left: 2rem;

  ${MediaQuery.medium`
    font-size: 1.5rem;
  `}
`;

export const HeaderWrapper = styled.div`
  display: flex;
  padding: 1.5rem 1.5rem 1.5rem 0;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid ${Colors.gray};

  ${MediaQuery.medium`
    padding: 1rem 1rem 1rem 0;
  `}
`;

export const NavList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

export const NavItem = styled.li`
  margin: 0;
  padding: 0;
  display: inline-block;

  a {
    color: ${Colors.black};
    padding: 1.5rem;
    text-transform: uppercase;
    &:hover {
      color: ${Colors.primary};
    }

    ${MediaQuery.medium`
      padding: 1rem;
  `}
  }
`;

const MainHeader = () => (
  <HeaderWrapper>
    <Branding>Elmwest</Branding>
    <nav>
      <NavList>
        <NavItem>
          <Link to="/">Home</Link>
        </NavItem>
        <NavItem>
          <Link to={'/'}>Shop</Link>
        </NavItem>
        <NavItem>
          <Link to={'/'}>Bedroom</Link>
        </NavItem>
      </NavList>
    </nav>
  </HeaderWrapper>
);

export default MainHeader;
