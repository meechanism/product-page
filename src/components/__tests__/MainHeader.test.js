import React from 'react';
import MainHeader, { Branding, NavItem, NavList } from '../MainHeader';

describe('MainHeader component', () => {
  it('should render', () => {
    const output = shallow(<MainHeader />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should contain branding', () => {
    const output = shallow(<MainHeader />);
    expect(output.find(Branding).exists()).toBeTruthy();
  });

  it('should contain nav menu', () => {
    const output = shallow(<MainHeader />);
    expect(output.find(NavList).exists()).toBeTruthy();
    expect(output.find(NavItem).length).toBe(3);
  });
});
