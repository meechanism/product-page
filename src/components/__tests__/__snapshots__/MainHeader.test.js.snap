// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader component should render 1`] = `
<styled.div>
  <styled.span>
    Elmwest
  </styled.span>
  <nav>
    <styled.ul>
      <styled.li>
        <Link
          to="/"
        >
          Home
        </Link>
      </styled.li>
      <styled.li>
        <Link
          to="/"
        >
          Shop
        </Link>
      </styled.li>
      <styled.li>
        <Link
          to="/"
        >
          Bedroom
        </Link>
      </styled.li>
    </styled.ul>
  </nav>
</styled.div>
`;
